src/network/networkd-iprule.c

```diff
@@ -0,0 +1,403 @@
+                if (r < 0)
+                        return log_error_errno(r, "Could not set ip rule table: %m");
+
+                r = sd_netlink_message_append_u32(m, FRA_TABLE, ip_rule->table);
+                if (r < 0)
+                        return log_error_errno(r, "Could not append FRA_TABLE attribute: %m");
+        }
+
+        if (ip_rule->fwmark > 0) {
+                r = sd_netlink_message_append_u32(m, FRA_FWMARK, ip_rule->fwmark);
+                if (r < 0)
+                        return log_error_errno(r, "Could not append FRA_FWMARK attribute:: %m");
+        }
+
+        if (ip_rule->fwmask > 0) {
+                r = sd_netlink_message_append_u32(m, FRA_FWMASK, ip_rule->fwmask);
+                if (r < 0)
+                        return log_error_errno(r, "Could not append FRA_FWMASK attribute:: %m");
+        }
+
+        r = sd_netlink_call_async(link->manager->rtnl, m, callback, link, 0, NULL);
+        if (r < 0)
+                return log_error_errno(r, "Could not send rtnetlink message: %m");
+
+        link_ref(link);
+
+        return 0;
+}
+
+int config_parse_ip_rule_tos(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_ip_rule_free_ IPRule *n = NULL;
+        Network *network = userdata;
+        uint32_t k;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = ip_rule_new_static(network, filename, section_line, &n);
+        if (r < 0)
+                return r;
+
+        r = safe_atou(rvalue, &k);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse ip rule tos, ignoring: %s", rvalue);
+                return 0;
+        }
+
+        n->tos = k;
+        n = NULL;
+
+        return 0;
+}
+
+int config_parse_ip_rule_table(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_ip_rule_free_ IPRule *n = NULL;
+        Network *network = userdata;
+        uint32_t k;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = ip_rule_new_static(network, filename, section_line, &n);
+        if (r < 0)
+                return r;
+
+        r = safe_atou(rvalue, &k);
```

Comment:
You can just use `&n->table` directly: safe_* function will not touch the output parameter unless successful.

There's a type mismatch here: `k` is `uint32_t`, but `safe_atou` takes an integer. You probably need to use `safe_atou32`. But you define `table` as unsigned in the header. So please decide what type is necessary and stick to it.